cmd/main.go

```diff
@@ -30,13 +32,32 @@ func Run(name string, args []string) error {
 	opts.args = flags.Args()
 	setupLogging(opts)
 
-	if opts.version {
+	switch {
+	case opts.version:
 		fmt.Fprintf(os.Stdout, "gotestsum version %s\n", version)
 		return nil
+	case opts.watch:
+		return runWatcher(opts)
 	}
 	return run(opts)
 }
 
+func runWatcher(opts *options) error {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Hour)
+	defer cancel()
+
+	fn := func(pkg string) error {
+		opts := *opts
+		opts.packages = []string{pkg}
+		err := run(&opts)
+		if ExitCodeWithDefault(err) > 1 {
+			return err
+		}
```

Comment:
Build errors exit with 2. This should only error if it's not an `exitCoder` error.